karyon3-archaius2/src/main/java/com/netflix/karyon/archaius/ArchaiusKaryonModule.java

```diff
@@ -133,109 +181,188 @@ public ArchaiusKaryonModule withCascadeStrategy(Class<? extends CascadeStrategy>
     
     @Override
     protected void configure() {
-        try {
-            if (appName != null) {
-                props.put(ServerContext.APP_ID, appName);
+        install(new ArchaiusAdminModule());
+        
+        bindListener(Matchers.any(), new ConfigurationInjectingListener());
+        bind(ConfigLifecycleListener.class).asEagerSingleton();
+        bind(CascadeStrategy.class).to(cascadeStrategy);
+        
+        for (String profile : profiles) {
+            if (!profile.isEmpty()) 
+                Multibinder.newSetBinder(binder(), String.class, Profiles.class).addBinding().toInstance(profile);
+        }
+        
+        Multibinder.newSetBinder(binder(), ConfigReader.class)
+            .addBinding().to(PropertiesConfigReader.class).in(Scopes.SINGLETON);
+        
+        MapBinder<String, Config> libraries = MapBinder.newMapBinder(binder(), String.class, Config.class, LibrariesLayer.class);
+        for (Map.Entry<String, Config> c : libraryOverrides.entrySet()) {
+            libraries.addBinding(c.getKey()).toInstance(c.getValue());
+        }
+        
+    }
+
+    @Provides
+    @Singleton
+    @RuntimeLayer
+    SettableConfig getSettableConfig() {
+        return runtimeLayer;
+    }
+    
+    @Provides
+    @Singleton
+    @ApplicationLayer 
+    CompositeConfig getApplicationLayer() {
+        return applicationLayer;
+    }
+
+    @Provides
+    @Singleton
+    @LibrariesLayer
+    CompositeConfig getLibrariesLayer() {
+        return librariesLayer;
+    }
+    
+    @Provides
+    @Singleton
+    PropertySource getPropertySource() {
+        return new AbstractPropertySource() {
+            @Override
+            public String get(String key) {
+                return rawConfig.getString(key, null);
             }
-            else if (getKaryon().getApplicationName() != null) {
-                props.put(ServerContext.APP_ID, getKaryon().getApplicationName());
+
+            @Override
+            public String get(String key, String defaultValue) {
+                return rawConfig.getString(key, defaultValue);
             }
+        };
+    }
+    
+    @Provides
+    @Singleton
+    @Raw
```

Comment:
I don't quite understand the raw annotation.  It's archaius config, but only pulled from the first 3 layers available locally rather than going to the network?